### PR TITLE
Issue #390

### PR DESCRIPTION
**Pull request by DamienBradleyDSP**
For #390

Small rework of human.nbt

-1 day  ->    1 day ago
-2 hours -> 2 hours ago

### CODE REVIEW

**Comment by sharkdp** [comment]
Thank you. From a quick glance, this seems to do more than just solve the problem with negative times. If so, can we please add regression tests for these other features/issues?

**Comment by mhmdanas** [comment]
> I assume you mean just expand the number of unit tests that are already there to something more comprehensive?

Yes, that's generally good practice and what we expect people to do when contributing features to Numbat. :)

File: examples/datetime_human_tests.nbt [hunk 1]
```diff
@@ -32,7 +51,46 @@ assert((1e12 days        -> human) == "1_000_000_000_000 days")
...
+assert((-10.5 decades    -> human) == "38350 days ago")
+
+assert((-25 ms           -> human) == "0.025 seconds ago")
+assert((-6400 ms         -> human) == "6.400 seconds ago")
+
+assert((-2500 µs         -> human) == "0.002500 seconds ago")
+assert((-40525 µs        -> human) == "0.040525 seconds ago")
+
+assert((-5147 ns         -> human) == "0.000005147 seconds ago")
+assert((-23s + 1234 ns   -> human) == "22.999998766 seconds ago")
```
Comment (sharkdp) [inline examples/datetime_human_tests.nbt:96]: The logic for "… ago" is relatively simple, right? In that case, we don't need to repeat all these tests here.

Comment (sharkdp) [reply examples/datetime_human_tests.nbt:96]: > True but does protect against changes if someone switches the logic in some way in the future, and with the test base's overall speed I'd say it doesn't hurt anyone to keep these in

Speed is not the matter, but humans are going to read those tests and wonder what they are for. I would really prefer if we strip down the negative-time-duration tests to one or two.

File: numbat/modules/datetime/human.nbt [hunk 2]
```diff
@@ -3,34 +3,51 @@ use core::strings
...
-fn _human_readable_duration(time: Time, dt: DateTime, num_days: Scalar) -> String =
-    _human_join(_human_join(_human_join(_human_days(_human_num_days(time)), _human_hours(dt)), _human_minutes(dt)), _human_seconds(dt))
+fn _human_readable_duration(time: Time) -> String =
+    _human_join(_human_join(_human_join(_human_formatted_days(time), _human_formatted_hours(time)), _human_formatted_minutes(time)), _human_formatted_seconds(time))
 
 # Implementation details:
 # we skip hours/minutes/seconds for durations larger than 1000 days because:
 #   (a) we run into floating point precision problems at the nanosecond level at this point
 #   (b) for much larger numbers, we can't convert to DateTimes anymore
+fn human_time_check_overflow(time: Time) =
```
Comment (sharkdp) [inline numbat/modules/datetime/human.nbt:45]: can we please `_`-prefix this as well?

**Comment by sharkdp** [comment]
I am still not sure what this PR brings in addition to the "… ago" handling of negative durations. For now, I'm going to close this in favor of #425. But please feel free to comment here and we can reopen it.

**Pull request closed without merging**